pkg/kube/controllers/boshdeployment/reconciler.go

```diff
@@ -135,24 +475,299 @@ func (r *ReconcileBOSHDeployment) Reconcile(request reconcile.Request) (reconcil
+			return errors.Wrap(err, "couldn't set reference for an ExtendedStatefulSet for a BOSH Deployment")
 		}
 
 		// Check to see if the object already exists
 		existingESts := &estsv1.ExtendedStatefulSet{}
-		err = r.client.Get(ctx, types.NamespacedName{Name: eSts.Name, Namespace: eSts.Namespace}, existingESts)
+		err := r.client.Get(ctx, types.NamespacedName{Name: eSts.Name, Namespace: eSts.Namespace}, existingESts)
 		if err != nil && apierrors.IsNotFound(err) {
-			r.log.Infof("Creating a new ExtendedStatefulset %s/%s for Deployment Manifest %s\n", eSts.Namespace, eSts.Name, instance.Name)
+			r.log.Infof("Creating a new ExtendedStatefulSet %s/%s for Deployment Manifest %s\n", eSts.Namespace, eSts.Name, instance.Name)
 
 			// Create the extended statefulset
 			err := r.client.Create(ctx, &eSts)
 			if err != nil {
 				r.recorder.Event(instance, corev1.EventTypeWarning, "CreateExtendedStatefulSetForDeployment Error", err.Error())
-				r.log.Errorf("Error creating ExtendedStatefulSet %s for deployment manifest %s: %s", eSts.Name, request.NamespacedName, err)
-				return reconcile.Result{}, errors.Wrap(err, "couldn't create an ExtendedStatefulSet for a BOSH Deployment")
+				r.log.Errorf("Error creating ExtendedStatefulSet %s for deployment manifest %s: %s", eSts.Name, instance.GetName(), err)
+				return errors.Wrap(err, "couldn't create an ExtendedStatefulSet for a BOSH Deployment")
 			}
 		}
 	}
 
-	return reconcile.Result{}, nil
+	instance.Status.State = deployingState
+
+	return nil
+}
+
+// actionOnDeploying check out deployment status
+func (r *ReconcileBOSHDeployment) actionOnDeploying(ctx context.Context, instance *bdc.BOSHDeployment, kubeConfigs *bdm.KubeConfig) error {
+	// TODO Check deployment
+	instance.Status.State = deployedState
+
+	return nil
+}
+
+// newExtendedJobTemplateForVariableInterpolation returns a job to interpolate variables
+func (r *ReconcileBOSHDeployment) newExtendedJobTemplateForVariableInterpolation(ctx context.Context, manifest *corev1.Secret, variables []esv1.ExtendedSecret, jobLabels map[string]string, secretLabels map[string]string, namespace string) (*ejv1.ExtendedJob, error) {
+	cmd := []string{"/bin/sh"}
+	args := []string{"-c", "cf-operator variable-interpolation --manifest /var/run/secrets/manifest/manifest.yaml --variables-dir /var/run/secrets/variables | base64 | tr -d '\n' | echo \"{\\\"interpolated-manifest.yaml\\\":\\\"$(</dev/stdin)\\\"}\""}
+
+	volumes := []corev1.Volume{
+		{
+			Name: string(manifest.GetUID()),
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: manifest.GetName(),
+					Items: []corev1.KeyToPath{
+						{
+							Key:  "manifest.yaml",
+							Path: "manifest.yaml",
+						},
+					},
+				},
+			},
+		},
+	}
+	volumeMounts := []corev1.VolumeMount{
+		{
+			Name:      string(manifest.GetUID()),
+			MountPath: "/var/run/secrets/manifest",
```

Comment:
No need for the double "manifest" - `manifest/manifest.yaml`.
Can you mount directly to `/var/run/secrets`?